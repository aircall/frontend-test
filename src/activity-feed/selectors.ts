import { selector } from "recoil";
import memoize from "lodash.memoize";

import { listState } from "../common/state/activity";

type ActivitiesIdsByDay = {
  day: string;
  activityIds: number[];
}[];

/**
 * Get activity IDs grouped by day and sort from the most recent to the least recent
 */
export const activityIdsGroupedByDayState = selector({
  key: "activityIdsGroupedByDayState",
  get: ({ get }) => {
    const list = get(listState);

    return list
      .filter((activity) => !activity.is_archived)
      .sort(({ created_at: created_at1 }, { created_at: created_at2 }) =>
        created_at1 < created_at2 ? 1 : -1
      )
      .map((a) => {
        const day = new Date(a.created_at);
        day.setHours(0, 0, 0, 0);
        return { day: day.toISOString(), a };
      })
      .reduce((list: ActivitiesIdsByDay, currentDay) => {
        const day = list.find((d) => d.day === currentDay.day);
        if (day) {
          day.activityIds.push(currentDay.a.id);
        } else {
          list.push({ day: currentDay.day, activityIds: [currentDay.a.id] });
        }
        return list;
      }, []);
  },
});

/**
 * Get one activity in the list by its ID
 */
export const activityByIdState = memoize((id: number) =>
  selector({
    key: `activityByIdState_${id}`,
    get: ({ get }) => {
      const list = get(listState);

      return list.find((activity) => activity.id === id);
    },
  })
);
