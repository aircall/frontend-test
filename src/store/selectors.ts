import { State } from './reducer';
import format from 'date-fns/format';

export function getStatus(state: State) {
  return state.calls.status;
}

export function getError(state: State) {
  return state.calls.error;
}

export function getFeed(state: State) {
  const { ids, entities } = state.calls;
  // Add creation_date property
  const callsWithDate = ids.map((id) => {
    const callById = entities[id];
    return {
      ...callById,
      creation_date: format(new Date(callById.created_at), 'MMMM,dd yyyy'),
    };
  });
  // filter unArchived calls
  const unArchivedCalls = callsWithDate.filter((c) => !c.is_archived);

  // get call's id by date
  let idsByDate: { [key: string]: { ids: string[]; date: string } } = {};
  idsByDate = unArchivedCalls.reduce((acc, cur) => {
    if (acc[cur.creation_date]) {
      acc[cur.creation_date].ids.push(cur.id);
    } else {
      acc[cur.creation_date] = { ids: [cur.id], date: cur.creation_date };
    }
    return acc;
  }, idsByDate);
  return idsByDate;
}

export function getCallById(state: State, id: string) {
  return state.calls.entities[id];
}
