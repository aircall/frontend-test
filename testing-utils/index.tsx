import React from "react";
import "jest-styled-components";
import { render, RenderOptions } from "@testing-library/react";
import { ThemeProvider } from "styled-components";

import { theme } from "../src/common/theme";

const customRender = (ui: React.ReactElement, options: RenderOptions = {}) => {
  const AllTheProviders = ({ children }: { children?: React.ReactNode }) => {
    return <ThemeProvider theme={theme}>{children}</ThemeProvider>;
  };

  return render(ui, { wrapper: AllTheProviders, ...options });
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render };
