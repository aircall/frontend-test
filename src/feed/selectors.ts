import { State } from '../store/reducer';
import format from 'date-fns/format';

export function getStatus(state: State) {
  return state.calls.status;
}

export function getError(state: State) {
  return state.calls.error;
}

export function getFeed(state: State) {
  const { ids, entities } = state.calls;
  const callsWithDate = ids.map((id) => {
    const callById = entities[id];
    return {
      ...callById,
      creation_date: format(new Date(callById.created_at), 'MMMM,dd yyyy'),
    };
  });
  let idsByDate: { [key: string]: { ids: string[]; date: string } } = {};
  idsByDate = callsWithDate.reduce((acc, cur) => {
    if (acc[cur.creation_date]) {
      acc[cur.creation_date].ids.push(cur.id);
    } else {
      acc[cur.creation_date] = { ids: [cur.id], date: cur.creation_date };
    }
    return acc;
  }, idsByDate);
  return idsByDate;
}

export function getCallById(state: State, id: string) {
  return state.calls.entities[id];
}
