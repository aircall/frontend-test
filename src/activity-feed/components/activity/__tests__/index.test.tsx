import React from "react";
import { useRecoilValue } from "recoil";

import { render } from "testing-utils";

import { Activity } from "..";
import { activityByIdState } from "../../../selectors";

jest.mock("recoil");
jest.mock("../../../selectors");

describe("activity-feed/components/activity", () => {
  describe("Activity", () => {
    describe("#render", () => {
      it("should render an anwsered outbound call correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "outbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "answered",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should render a missed outbound call correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "outbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "missed",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should render an anwsered inbound call correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "inbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "answered",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should render a missed inbound call correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "inbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "missed",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should render a missed outbound call with voicemail correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "outbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "voicemail",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should render a missed inbound call with voicemail correctly", () => {
        (useRecoilValue as jest.Mock).mockReturnValue({
          id: 7834,
          created_at: "2018-04-19T09:38:41.000Z",
          direction: "inbound",
          from: "Pierre-Baptiste Béchu",
          to: "06 46 62 12 33",
          via: "NYC Office",
          duration: "120",
          is_archived: false,
          call_type: "voicemail",
        });
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });

      it("should not render anything if the activity is undefined", () => {
        (useRecoilValue as jest.Mock).mockReturnValue(undefined);
        const { container } = render(<Activity id={42} />);

        expect(activityByIdState).toHaveBeenCalledWith(42);
        expect(container).toMatchSnapshot();
      });
    });
  });
});
