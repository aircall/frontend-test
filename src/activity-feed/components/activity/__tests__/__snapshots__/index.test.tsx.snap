// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`activity-feed/components/activity Activity #render should not render anything if the activity is undefined 1`] = `<div />`;

exports[`activity-feed/components/activity Activity #render should render a missed inbound call correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #e82300;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19.59 7L12 14.59 6.41 9H11V7H3v8h2v-4.59l7 7 9-9z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        Pierre-Baptiste Béchu
      </p>
      <p
        class="c7"
      >
        06 46 62 12 33
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;

exports[`activity-feed/components/activity Activity #render should render a missed inbound call with voicemail correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #05c1eb;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M19.59 7L12 14.59 6.41 9H11V7H3v8h2v-4.59l7 7 9-9z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        Pierre-Baptiste Béchu
      </p>
      <p
        class="c7"
      >
        06 46 62 12 33
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;

exports[`activity-feed/components/activity Activity #render should render a missed outbound call correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #e82300;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <path
            d="M24 24H0V0h24v24z"
          />
        </defs>
        <clippath>
          <use
            overflow="visible"
            xlink:href="#a"
          />
        </clippath>
        <path
          clip-path="url(#b)"
          d="M3 8.41l9 9 7-7V15h2V7h-8v2h4.59L12 14.59 4.41 7 3 8.41z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        06 46 62 12 33
      </p>
      <p
        class="c7"
      >
        Pierre-Baptiste Béchu
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;

exports[`activity-feed/components/activity Activity #render should render a missed outbound call with voicemail correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #05c1eb;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <defs>
          <path
            d="M24 24H0V0h24v24z"
          />
        </defs>
        <clippath>
          <use
            overflow="visible"
            xlink:href="#a"
          />
        </clippath>
        <path
          clip-path="url(#b)"
          d="M3 8.41l9 9 7-7V15h2V7h-8v2h4.59L12 14.59 4.41 7 3 8.41z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        06 46 62 12 33
      </p>
      <p
        class="c7"
      >
        Pierre-Baptiste Béchu
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;

exports[`activity-feed/components/activity Activity #render should render an anwsered inbound call correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #02c91c;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20 5.41L18.59 4 7 15.59V9H5v10h10v-2H8.41z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        Pierre-Baptiste Béchu
      </p>
      <p
        class="c7"
      >
        06 46 62 12 33
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;

exports[`activity-feed/components/activity Activity #render should render an anwsered outbound call correctly 1`] = `
.c2 {
  display: inline-block;
  vertical-align: middle;
  overflow: hidden;
}

.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  border: 0.0625rem solid #e5e5e5;
  padding: 0.75rem;
  border-radius: 0.25rem;
}

.c1 {
  position: relative;
}

.c3 {
  width: 2rem;
  color: #e5e5e5;
}

.c4 {
  position: absolute;
  width: 1rem;
  right: 0.2rem;
  top: 0.25rem;
  color: #02c91c;
}

.c5 {
  margin: 0 0.75rem;
  -webkit-box-flex: 1;
  -webkit-flex-grow: 1;
  -ms-flex-positive: 1;
  flex-grow: 1;
}

.c6 {
  margin: 0;
  font-size: 1rem;
  margin-bottom: 0.25rem;
  font-weight: bold;
}

.c7 {
  margin: 0;
  color: #b3b3b3;
}

<div>
  <section
    class="c0"
  >
    <span
      class="c1"
    >
      <svg
        aria-hidden="true"
        class="c2 c3"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M20.01 15.38c-1.23 0-2.42-.2-3.53-.56a.977.977 0 00-1.01.24l-1.57 1.97c-2.83-1.35-5.48-3.9-6.89-6.83l1.95-1.66c.27-.28.35-.67.24-1.02-.37-1.11-.56-2.3-.56-3.53 0-.54-.45-.99-.99-.99H4.19C3.65 3 3 3.24 3 3.99 3 13.28 10.73 21 20.01 21c.71 0 .99-.63.99-1.18v-3.45c0-.54-.45-.99-.99-.99z"
        />
      </svg>
      <svg
        aria-hidden="true"
        class="c2 c4"
        fill="currentColor"
        focusable="false"
        viewBox="0 0 24 24"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M0 0h24v24H0z"
          fill="none"
        />
        <path
          d="M9 5v2h6.59L4 18.59 5.41 20 17 8.41V15h2V5z"
        />
      </svg>
    </span>
    <section
      class="c5"
    >
      <p
        class="c6"
      >
        06 46 62 12 33
      </p>
      <p
        class="c7"
      >
        Pierre-Baptiste Béchu
      </p>
    </section>
    <span
      class=""
    >
      11:38 AM
    </span>
  </section>
</div>
`;
